### DISCUSSION TITLE
Firefox related nuisances

### DISCUSSION THREAD

**Discussion by dazu89** [simeononsecurity/Windows-Optimize-Harden-Debloat#48] [Q&A] [answered]
The following nuisances  related to Firefox I have been unable to resolve, if someone has a workaround, I would appreciate you reaching out. 
- Auto update of extensions [Registry Setting found, but change not persistent]  
- History is deleted [Registry Setting found, but change not persistent]  
- No prompt to allow use of microphone or camera [Setting not found]  
- Application Window size and position is not saved [Setting not found]  

What I have tried so far: Locate and modify the settings in the registry, but changes are overwritten after each restart; the mechanism is unclear to me. I have looked for the ADMX Policy definitions but could not find them neither in file nor via Group Policy Editor GUI, where they should show according to this [entry](https://learn.microsoft.com/en-us/windows/client-management/understanding-admx-backed-policies).

**Comment by simeononsecurity** [comment, marked as answer]
@dazu89 
So firefox can be a weird one. To satisfy some STIGs we had to configure them with FireFox configuration policies. Others were available via GPO so they were done there as well. 

If you go through the files [here](https://github.com/simeononsecurity/Windows-Optimize-Harden-Debloat/tree/master/Files/FireFox%20Configuration%20Files) you'll find the ones not available via GPO.

If you download the following repo and go [here](https://github.com/simeononsecurity/STIG-Compliant-Domain-Prep/tree/master/Files/PolicyDefinitions) you'll find the policy definitions that need to be imported to see the local GPOs set for FireFox. This is described [here](https://github.com/simeononsecurity/STIG-Compliant-Domain-Prep/tree/master/Files/PolicyDefinitions) on the readme of this repo.

**Comment by simeononsecurity** [comment]
If you find any ones you have issues with in the non gpo section, review the following script section to get he pathing. Delete the files that were copied the first time you ran the script. Modify the ones in your offline copy and then run the script again.
```#https://www.itsupportguides.com/knowledge-base/tech-tips-tricks/how-to-customise-firefox-installs-using-mozilla-cfg/
$firefox64 = "C:\Program Files\Mozilla Firefox"
$firefox32 = "C:\Program Files (x86)\Mozilla Firefox"
Write-Host "Installing Firefox Configurations - Please Wait." -ForegroundColor White -BackgroundColor Black
Write-Host "Window will close after install is complete" -ForegroundColor White -BackgroundColor Black
If (Test-Path -Path $firefox64) {
    Copy-Item -Path .\Files\"FireFox Configuration Files"\defaults -Destination $firefox64 -Force -Recurse
    Copy-Item -Path .\Files\"FireFox Configuration Files"\mozilla.cfg -Destination $firefox64 -Force
    Copy-Item -Path .\Files\"FireFox Configuration Files"\local-settings.js -Destination $firefox64 -Force 
    Write-Host "Firefox 64-Bit Configurations Installed" -ForegroundColor Green -BackgroundColor Black
}
Else {
    Write-Host "FireFox 64-Bit Is Not Installed" -ForegroundColor Red -BackgroundColor Black
}
If (Test-Path -Path $firefox32) {
    Copy-Item -Path .\Files\"FireFox Configuration Files"\defaults -Destination $firefox32 -Force -Recurse
    Copy-Item -Path .\Files\"FireFox Configuration Files"\mozilla.cfg -Destination $firefox32 -Force
    Copy-Item -Path .\Files\"FireFox Configuration Files"\local-settings.js -Destination $firefox32 -Force 
    Write-Host "Firefox 32-Bit Configurations Installed" -ForegroundColor Green -BackgroundColor Black
}
Else {
    Write-Host "FireFox 32-Bit Is Not Installed" -ForegroundColor Red -BackgroundColor Black
}```

**Comment by dazu89** [comment]
Thank you so much, @simeononsecurity, for your help! Indeed, importing \*.admx and \*.adml  to the folder `C:\Windows\PolicyDefinitions` and `C:\Windows\PolicyDefinitions\en-US\` respectively made polices show up in the *group policy editor*. I have not found a Powershell library to conveniently manage those entries locally.

Maybe someone else stumbles over these firefox issues:
- Auto update of extensions [modify via GPO]
- History is deleted [modify via GPO]
- No prompt to allow use of microphone or camera [modify via mozilla.cfg; set media.navigator.enabled to true]
- Application Window size and position is not saved [modify via mozilla.cfg; set privacy.resistFingerprinting to false]

**Reply by simeononsecurity** [reply]
We've found one powershell script that does, kind of... But it wouldn't work for this use case...
Normally we just gather those manually. There isn't a good way to manage them other than manually that we're aware of unfortunately.
